procdiscovery/pkg/process/process.go

```diff
@@ -130,10 +144,21 @@ func getRelevantEnvVars(pid int) map[string]string {
 			continue
 		}
 
-		if _, ok := relevantEnvVars[envParts[0]]; ok {
-			result[envParts[0]] = envParts[1]
+		if _, ok := relevantOverwriteEnvVars[envParts[0]]; ok {
+			overWriteEnvsResult[envParts[0]] = envParts[1]
 		}
+
+		for _, env := range envDetailsSeparatorMap {
```

Comment:
Consider converting `envDetailsSeparatorMap` from slice to `map[strings]struct{}`
This way you can look directly the keys instead of iterating the map every time